### PR TITLE
Record structs have no copy constructors

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/54413

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs [hunk 1]
```diff
@@ -10568,5 +10568,38 @@ public void BaseArguments_Speculation()
...
         }
+
+        [Fact, WorkItem(54413, "https://github.com/dotnet/roslyn/issues/54413")]
+        public void Repro54413()
+        {
+            var src = @"
+record struct Value(string Text)
+{
+    private Value(int X) { } // 1
+    private Value(Value original) { } // 2
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs:10579]: I assume that before this change, diagnostic '2' and '4' were missing?

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs:10579]: Diagnostic 2 was missing.
In record classes, copy constructors are special (ie. the concept exists in the spec and has rules around it). That's why we have diagnostic 4, which is different than diagnostic 2.

Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs:10579]: Thanks, I missed that the record below is a record class.￼

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs:10579]: Do we have a passing test with a `record struct` with a constructor that has the signature of a copy constructor?

File: src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs [hunk 2]
```diff
@@ -10568,5 +10568,38 @@ public void BaseArguments_Speculation()
 
             Assert.False(model.TryGetSpeculativeSemanticModel(baseWithoutParens.SpanStart + 2, speculativeBase, out _));
         }
+
+        [Fact, WorkItem(54413, "https://github.com/dotnet/roslyn/issues/54413")]
+        public void Repro54413()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs:10573]: Consider giving a descriptive name, perhaps "ValueTypeConstructor_NoThisInitializer".

**Review by cston — APPROVED** [review]

**Pull request merged**